apollo-router/src/plugin/mod.rs

```diff
@@ -299,16 +310,13 @@ macro_rules! register_plugin {
     ($group: literal, $name: literal, $plugin_type: ident) => {
         //  Artificial scope to avoid naming collisions
         const _: () = {
-            #[$crate::_private::ctor::ctor]
-            fn register_plugin() {
-                let qualified_name = if $group == "" {
-                    $name.to_string()
-                } else {
-                    format!("{}.{}", $group, $name)
-                };
-
-                $crate::plugin::register_plugin::<$plugin_type>(qualified_name);
-            }
+            use $crate::_private::once_cell::sync::Lazy;
+            use $crate::_private::PluginFactory;
+            use $crate::_private::PLUGINS;
+
+            #[$crate::_private::linkme::distributed_slice(PLUGINS)]
```

Comment:
Although [that feature is undocumented](https://github.com/dtolnay/linkme/blob/0.3.6/impl/src/attr.rs#L4), I’ve tested that this change removes the need for plugin crates to have their own dependency on `linkme`, making this PR a non-breaking change:

```suggestion
            #[$crate::_private::linkme::distributed_slice(PLUGINS)]
            #[linkme(crate = $crate::_private::linkme)]
```

Reply:
Great suggestion! Thank you.

Reply:
done: https://github.com/apollographql/router/pull/2181/commits/cf3b62b33fe674d8cdc307cd2a40f9dc33371dc0